src/python/WMCore/Database/MongoDB.py

```diff
@@ -156,7 +161,8 @@ def _dbConnect(self, db):
         """
         try:
             setattr(self, db, self.client[db])
-            self._dbTest(db)
+            if not self.mockMongoDB:
```

Comment:
Todor, I'd suggest to move this check for `self.mockMongoDB` to the first line(s) in the `_dbTest` method.

Reply:
Actually, this would make the code less readable. The `_dbTest`  method has no return statement, it is acting with the database with a `command` that is not supported by mongomock [1]. And actually we do not need to test the database at all while working with `mongomock`, so I'd prefer to not even call the method.  


[1]
```
NotImplementedError: command is a valid Database method but is not implemented in Mongomock yet  
```

Reply:
Yes, hence my suggestion to check it at the very first line of that method, to skip it when it's not actually needed (it's being emulated). 
To me, having that check at the very beginning of the method that is supposed to act differently according to the real vs emulation mode would make it more readable. This nested try/except block makes it not a the best spot IMO.
Code is still fully functional though, so if you feel like keeping it like that, let's keep it.

Reply:
Agreed, but the method that is supposed to act differently based on the `mockMongoDb` flag is not `_dbTest`, but rather `_dbConnect`, because we are not actually connecting to any database we are mocking it instead.

Reply:
I am suggesting an alternative to the code you are proposing. If you say it's only `_dbConnect` that should not be called when mocking MongoDB, then it's probably better to update this code, no?

Reply:
I said  `_dbConnect` should behave differently when called with and without mocking. But it still needs to be called, because it sets the database attribute to the object on line 163. This is an attribute of the objects that  other pieces of the code rely on. The difference is that when mocking the database then no database test is needed, so no need to call the `_dbTest` method. I am sure this is the right place for this check.

Reply:
Then the `_dbConnect` comment is meaningless for what we are discussing here, which is what the best place is for checking whether mongodb is being mocked or not. Anyhow, it's time to move on.